lambdas/certexpiryreminder/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/handler/CertExpiryReminderHandler.java

```diff
@@ -0,0 +1,110 @@
+import static uk.gov.di.ipv.cri.drivingpermit.library.metrics.Definitions.CERTIFICATE_EXPIRY_REMINDER;
+
+public class CertExpiryReminderHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    public static final String AUTHORIZATION_HEADER_KEY = "Authorization";
+    public static final String LAMBDA_HANDLING_EXCEPTION =
+            "Exception while handling lambda {} exception {}";
+    private final Region awsRegion = Region.of(System.getenv("AWS_REGION"));
+
+    public CertExpiryReminderHandler() {}
+
+    @Override
+    @Logging(correlationIdPath = CorrelationIdPathConstants.API_GATEWAY_REST)
+    @Metrics(captureColdStart = true)
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+
+        ConfigurationService configurationService = null;
+        try {
+            configurationService = new ConfigurationService();
+        } catch (CertificateException | NoSuchAlgorithmException | InvalidKeySpecException e) {
+            throw new RuntimeException(e);
+        }
+
+        Date signingCertExpiry =
+                ((X509Certificate) configurationService.getDvaConfiguration().getSigningCert())
+                        .getNotAfter();
+        Date encryptionCertExpiry =
+                ((X509Certificate) configurationService.getDvaConfiguration().getEncryptionCert())
+                        .getNotAfter();
+        Date tlsRootCertExpiry =
+                ((X509Certificate) configurationService.getDvaConfiguration().getTlsRootCert())
+                        .getNotAfter();
+        Date tlsIntermediateCertExpiry =
+                ((X509Certificate)
+                                configurationService.getDvaConfiguration().getTlsIntermediateCert())
+                        .getNotAfter();
+        Date tlsCertExpiry =
+                ((X509Certificate) configurationService.getDvaConfiguration().getTlsSelfCert())
+                        .getNotAfter();
+
+        ArrayList<LocalDate> certExpirys = new ArrayList<LocalDate>();
+        certExpirys.add(convertToLocalDate(signingCertExpiry));
+        certExpirys.add(convertToLocalDate(encryptionCertExpiry));
+        certExpirys.add(convertToLocalDate(tlsRootCertExpiry));
+        certExpirys.add(convertToLocalDate(tlsIntermediateCertExpiry));
+        certExpirys.add(convertToLocalDate(tlsCertExpiry));
+
+        ArrayList<String> certPaths = new ArrayList<String>();
+        certPaths.add(DVA_DRIVING_PERMIT_CRI_SIGNING_CERT);
+        certPaths.add(DVA_ENCRYPTION_CERT);
+        certPaths.add(DVA_HTTPCLIENT_TLS_ROOT_CERT);
+        certPaths.add(DVA_HTTPCLIENT_TLS_INTER_CERT);
+        certPaths.add(DVA_HTTPCLIENT_TLS_CERT);
+
+        LocalDate expiryWindow = LocalDate.now().minusWeeks(2);
+
+        for (int i = 0;
+                certExpirys.get(i).isAfter(expiryWindow)
```

Comment:
Do these conditions not need switched so its certs expiring between now and the window?